### ISSUE TITLE
ffmetrics doesn't work on 16 bit ffv1 .mkv file

### ISSUE THREAD

**Issue by JohnTravolski** [fifonik/FFMetrics#154]
If I encode a 16-bpc png sequence using this ffmpeg command:

```
ffmpeg -framerate 60 -i "out_%05d.png" -c:v ffv1 -level 3 -coder 1 -context 1 -pix_fmt gbrp16le "out.mkv"
```
and then I use this file as the reference in ffmetrics, the "Start" button remains grayed out forever. It works if I use a yuv pixel format like this instead:

```
ffmpeg -framerate 60 -i "out_%05d.png" -c:v libx264 -preset veryslow -qp 0 -pix_fmt yuv444p16le "out.mkv"
```

Does ffmetrics not support gbrp16le? Or does it not support ffv1? I'm not sure why it won't work.

**Comment by fifonik** [comment]
I will try to re-produce it. No idea yet what is going on with such format. It is possible that my code that parsing media info cannot handle this.

**Comment by fifonik** [comment]
Converted short file using command:
`ffmpeg -i 2012-11-22_095841.m2ts -c:v ffv1 -level 3 -coder 1 -context 1 -pix_fmt gbrp16le 2012-11-22_095841_gbrp16le.mkv`
And all good -- FFMetrics 1.6b8 parsed it media info and compared it with original.
What FFMetrics version you use?

**Comment by fifonik** [comment]
My file was smaller (~5GB), but I created bigger file (~52GB) and everything is good as well:
![image](https://github.com/user-attachments/assets/d6a95cc5-1247-43ad-9c6e-2225f35d4977)
```
Media Info:
General
Unique ID                                : 66197689196928693206931944942073612216 (0x31CD36105FF66B4A82FCCC82901D07B8)
Complete name                            : J:\Temp\gbrp16le.mkv
Format                                   : Matroska
Format version                           : Version 4
File size                                : 52.1 GiB
Duration                                 : 2 min 37 s
Overall bit rate mode                    : Variable
Overall bit rate                         : 2 833 Mb/s
Frame rate                               : 50.000 FPS
Writing application                      : Lavf58.76.100
Writing library                          : Lavf58.76.100
ErrorDetectionType                       : Per level 1

Video
ID                                       : 1
Format                                   : FFV1
Format version                           : Version 3.4
Codec ID                                 : V_MS/VFW/FOURCC / FFV1
Duration                                 : 2 min 37 s
Bit rate mode                            : Variable
Bit rate                                 : 2 776 Mb/s
Width                                    : 1 920 pixels
Height                                   : 1 080 pixels
Display aspect ratio                     : 16:9
Frame rate mode                          : Constant
Frame rate                               : 50.000 FPS
Color space                              : RGB
Bit depth                                : 16 bits
Scan type                                : Progressive
Compression mode                         : Lossless
Bits/(Pixel*Frame)                       : 26.778
Stream size                              : 51.0 GiB (98%)
Writing library                          : Lavc58.134.100 ffv1
Default                                  : Yes
Forced                                   : No
Color range                              : Full
coder_type                               : Range Coder
MaxSlicesCount                           : 4
ErrorDetectionType                       : Per slice
```

Based on your screenshot, I think the issue is not with the encoded file, but with reference instead as FFMetrics does not show any media info for it.

**Comment by fifonik** [comment]
You can try to open the with with FFMpeg in a way FFMetrics do it to get media info:
```
ffmpeg -hide_banner -nostdin -t 00:00:02 -i out.mkv -frames:v 30 -filter:v idet -f null -
```

**Comment by JohnTravolski** [comment]
I don't see anything in `FFMetrics.log.` Running your line above on the .mkv file gave me:

```
ffmpeg -t 00:00:02 -i wavyheart.mkv -frames:v 30 -filter:v idet -f null -

ffmpeg version 2024-10-10-git-0f5592cfc7-full_build-www.gyan.dev Copyright (c) 2000-2024 the FFmpeg developers
  built with gcc 14.2.0 (Rev1, Built by MSYS2 project)
  configuration: --enable-gpl --enable-version3 --enable-static --disable-w32threads --disable-autodetect --enable-fontconfig --enable-iconv --enable-gnutls --enable-libxml2 --enable-gmp --enable-bzlib --enable-lzma --enable-libsnappy --enable-zlib --enable-librist --enable-libsrt --enable-libssh --enable-libzmq --enable-avisynth --enable-libbluray --enable-libcaca --enable-sdl2 --enable-libaribb24 --enable-libaribcaption --enable-libdav1d --enable-libdavs2 --enable-libopenjpeg --enable-libquirc --enable-libuavs3d --enable-libxevd --enable-libzvbi --enable-libqrencode --enable-librav1e --enable-libsvtav1 --enable-libvvenc --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxavs2 --enable-libxeve --enable-libxvid --enable-libaom --enable-libjxl --enable-libvpx --enable-mediafoundation --enable-libass --enable-frei0r --enable-libfreetype --enable-libfribidi --enable-libharfbuzz --enable-liblensfun --enable-libvidstab --enable-libvmaf --enable-libzimg --enable-amf --enable-cuda-llvm --enable-cuvid --enable-dxva2 --enable-d3d11va --enable-d3d12va --enable-ffnvcodec --enable-libvpl --enable-nvdec --enable-nvenc --enable-vaapi --enable-libshaderc --enable-vulkan --enable-libplacebo --enable-opencl --enable-libcdio --enable-libgme --enable-libmodplug --enable-libopenmpt --enable-libopencore-amrwb --enable-libmp3lame --enable-libshine --enable-libtheora --enable-libtwolame --enable-libvo-amrwbenc --enable-libcodec2 --enable-libilbc --enable-libgsm --enable-liblc3 --enable-libopencore-amrnb --enable-libopus --enable-libspeex --enable-libvorbis --enable-ladspa --enable-libbs2b --enable-libflite --enable-libmysofa --enable-librubberband --enable-libsoxr --enable-chromaprint
  libavutil      59. 42.100 / 59. 42.100
  libavcodec     61. 22.100 / 61. 22.100
  libavformat    61.  9.100 / 61.  9.100
  libavdevice    61.  4.100 / 61.  4.100
  libavfilter    10.  6.100 / 10.  6.100
  libswscale      8.  5.100 /  8.  5.100
  libswresample   5.  4.100 /  5.  4.100
  libpostproc    58.  4.100 / 58.  4.100
Input #0, matroska,webm, from 'wavyheart.mkv':
  Metadata:
    ENCODER         : Lavf61.9.100
  Duration: 00:00:28.00, start: 0.000000, bitrate: 15074363 kb/s
  Stream #0:0: Video: ffv1 (FFV1 / 0x31564646), gbrp16le(pc, gbr/unknown/unknown, progressive), 3840x2160, SAR 1:1 DAR 16:9, 60 fps, 60 tbr, 1k tbn
    Metadata:
      ENCODER         : Lavc61.22.100 ffv1
      DURATION        : 00:00:28.000000000
[Parsed_idet_0 @ 00000154cc507f40] Repeated Fields: Neither:     0 Top:     0 Bottom:     0
[Parsed_idet_0 @ 00000154cc507f40] Single frame detection: TFF:     0 BFF:     0 Progressive:     0 Undetermined:     0
[Parsed_idet_0 @ 00000154cc507f40] Multi frame detection: TFF:     0 BFF:     0 Progressive:     0 Undetermined:     0
Stream mapping:
  Stream #0:0 -> #0:0 (ffv1 (native) -> wrapped_avframe (native))
Press [q] to stop, [?] for help
Output #0, null, to 'pipe:':
  Metadata:
    encoder         : Lavf61.9.100
  Stream #0:0: Video: wrapped_avframe, yuv444p16le(progressive), 3840x2160 [SAR 1:1 DAR 16:9], q=2-31, 200 kb/s, 60 fps, 60 tbn
    Metadata:
      encoder         : Lavc61.22.100 wrapped_avframe
      DURATION        : 00:00:28.000000000
[Parsed_idet_0 @ 00000154cca4ec80] Repeated Fields: Neither:    30 Top:     0 Bottom:     0
[Parsed_idet_0 @ 00000154cca4ec80] Single frame detection: TFF:     0 BFF:     0 Progressive:    30 Undetermined:     0
[Parsed_idet_0 @ 00000154cca4ec80] Multi frame detection: TFF:     0 BFF:     0 Progressive:    30 Undetermined:     0
[out#0/null @ 00000154cc46b140] video:13KiB audio:0KiB subtitle:0KiB other streams:0KiB global headers:0KiB muxing overhead: unknown
frame=   30 fps=3.6 q=-0.0 Lsize=N/A time=00:00:00.50 bitrate=N/A speed=0.0595x
```
Looks like it worked to me. So I'll have to do some more digging later when I get time.

**Comment by fifonik** [comment]
And how long did it take? 
In your output it shows **fps=3.6** and the command asks ffmpeg to analyze 30 frames fo theexecution time should be about 30/3.6=8.2 seconds. 
This is lower than the operation timeout (10 seconds as I mentioned earlier). 
However, it might be some overheads and so the full process time might take more than 10 seconds so timeout.
I would expect to see error message about that in FFMetrics.log (the log only created if FFMetrics have permissions to write near executable).
To check you can try to increase the timeout by creating FFMetrics.conf with the following content:
```
{
	"MediaInfo": {
		"Timeout": 30000
	}
}
```

**Comment by JohnTravolski** [comment]
That appears to have been it. I just tried 1.5b10 and it works now! It did take a while for it to load the .mkv (somewhere between 15 and 20 seconds).

![image](https://github.com/user-attachments/assets/a4929feb-b642-4d76-930a-4cd4ea525043)

Thanks!

As for why no timeout errors were logged, I can't say why; I have it installed in a custom directory where I have a lot of other portable programs installed (not C:\Program Files) so it should have full permissions.

**Comment by fifonik** [comment]
> As for why no timeout errors were logged, I can't say why; I have it installed in a custom directory where I have a lot of other portable programs installed (not C:\Program Files) so it should have full permissions.

Thanks for the report, I'll double check

**Comment by fifonik** [comment]
I double checked and confirming that timeout errors were only logged if program started with parameter `-log-level=debug`
Will fix it shortly.

**Comment by fifonik** [comment]
FYI: I've fixed timeouts not logged issue in 1.5.0 beta 11
